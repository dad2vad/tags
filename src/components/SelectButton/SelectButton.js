import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Icon from '../Icon';

class SelectButton extends PureComponent {
  buttonClass = () =>
    classNames('button', 'is-checkbox', 'is-info', {
      'is-outlined': !this.props.isSelected,
    });

  iconClass = () =>
    classNames({
      'check-square': this.props.isSelected,
      square: !this.props.isSelected,
    });

  render() {
    return (
      <button
        onClick={() => {
          this.props.onClick(this.props.framework.slug);
        }}
        className={this.buttonClass()}
      >
        <Icon name={this.iconClass()} />
        <span>{this.props.framework.name}</span>
        {this.props.isNew && <span className="tag is-warning">New</span>}
      </button>
    );
  }
}

SelectButton.propTypes = {
  framework: PropTypes.shape({
    slug: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
  }),
  onClick: PropTypes.func.isRequired,
  isNew: PropTypes.bool.isRequired,
  isSelected: PropTypes.bool.isRequired,
};

SelectButton.defaultProps = {
  isNew: false,
  isSelected: false,
};

export default SelectButton;
