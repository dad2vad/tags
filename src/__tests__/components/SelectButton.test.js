import React from 'react';
import { shallow } from 'enzyme';
import SelectButton from '../../components/SelectButton';

describe('Select Button', () => {
  it('renders correctly', () => {
    const component = shallow(
      <SelectButton
        onClick={() => {}}
        isSelected={false}
        framework={{ name: 'Foo', slug: 'bar' }}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
