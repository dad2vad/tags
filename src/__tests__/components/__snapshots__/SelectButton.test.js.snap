// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Button renders correctly 1`] = `
<button
  className="button is-checkbox is-info is-outlined"
  onClick={[Function]}
>
  <Icon
    name="square"
  />
  <span>
    Foo
  </span>
</button>
`;
